src/inspector_agent.cc

```diff
@@ -387,29 +408,29 @@ bool Agent::Start(v8::Platform* platform, const char* path,
           new NodeInspectorClient(parent_env_, platform));
   inspector_->contextCreated(parent_env_->context(), "Node.js Main Context");
   platform_ = platform;
+  CHECK_EQ(0, uv_async_init(uv_default_loop(),
+                            &start_inspector_thread_async,
+                            StartInspectorIoThreadAsyncCallback));
+  start_inspector_thread_async.data = this;
+  uv_unref(reinterpret_cast<uv_handle_t*>(&start_inspector_thread_async));
+
+  RegisterDebugSignalHandler();
```

Comment:
I think the reason for moving this out of the if/else clause is that since `_debugEnd` can stop the running socket server, we need to register the signal handler even when `--inspect` was specified, so that after the socket server is stopped it can still be restarted.

Makes sense to me. 👍